ui/src/infra-model/view/infra-model-view-container.tsx

```diff
@@ -37,10 +47,23 @@ function mapDispatchToProps(dispatch: TrackLayoutAppDispatch) {
         onHoverLocation: delegates.onHoverLocation,
         onClickLocation: delegates.onClickLocation,
         onViewportChange: delegates.onViewportChange,
-        onInfraModelViewChange: delegates.onViewChange,
         onCommitField: delegates.onCommitField,
         showArea: delegates.showArea,
+        setExistingInfraModel: delegates.setExistingInfraModel,
     };
 }
 
-export const InfraModelViewContainer = connect(mapStateToProps, mapDispatchToProps)(InfraModelView);
+const InfraModelLoadingInfraModelView: React.FC<InfraModelViewProps> = (
+    props: InfraModelLoaderProps,
+) => {
+    return props.viewType == InfraModelViewType.UPLOAD ? (
+        <InfraModelView {...props} />
+    ) : (
+        <InfraModelEditLoader {...props} />
+    );
```

Comment:
Jäin miettimään että kannattaisiko tämä iffittelylogiikka siirtää osaksi `InfraModelEditLoader`:ia, sillä nyt tästä ja sen päätason iffeistä muodostuu vähän niinkuin 3 polkua, joista 2 vie `InfraModelView`:iin ja yksi `Spinner`:iin. Kuulostaisi siltä, että nuo viewiin vievät polut olisi ehkä mahdollista yhdistää jotenkin yhden ehdon alle.